Use the same Python interpreter that the passed-in Meson does.
Fix PowerPC.

--- configure.orig	2023-08-11 06:51:47.000000000 +0800
+++ configure	2023-08-18 18:48:53.000000000 +0800
@@ -519,6 +519,10 @@
   else
     cpu="sparc"
   fi
+elif check_define __ppc64__ ; then
+  cpu="ppc64"
+elif check_define __ppc__ ; then
+  cpu="ppc"
 elif check_define _ARCH_PPC ; then
   if check_define _ARCH_PPC64 ; then
     if check_define _LITTLE_ENDIAN ; then
@@ -577,9 +581,9 @@
     cpu="mips" ;;
 
   ppc)
-    CPU_CFLAGS="-m32" ;;
+    CPU_CFLAGS="-arch ppc" ;;
   ppc64)
-    CPU_CFLAGS="-m64 -mbig-endian" ;;
+    CPU_CFLAGS="-arch ppc64 -mcpu970" ;;
   ppc64le)
     cpu="ppc64"
     CPU_CFLAGS="-m64 -mlittle-endian" ;;
@@ -1058,6 +1062,13 @@
 # Remove old dependency files to make sure that they get properly regenerated
 rm -f ./*/config-devices.mak.d
 
+# If --meson is supplied but --python is not, extract the Python interpreter
+# from the Meson script
+if test "$explicit_python" = no && ! test -z "$meson" && test "$meson" != "git" && test "$meson" != "internal"
+then
+    python=$(sed -n '1s/^#!//p' $(command -v "$meson"))
+fi
+
 if test -z "$python"
 then
     # If first_python is set, there was a binary somewhere even though
@@ -1916,29 +1927,29 @@
 
   target_arch=${1%%-*}
   case $target_arch in
-    aarch64) container_hosts="x86_64 aarch64" ;;
+    aarch64) container_hosts="x86_64 aarch64 ppc ppc64" ;;
     alpha) container_hosts=x86_64 ;;
-    arm) container_hosts="x86_64 aarch64" ;;
+    arm) container_hosts="x86_64 aarch64 ppc ppc64" ;;
     cris) container_hosts=x86_64 ;;
     hexagon) container_hosts=x86_64 ;;
     hppa) container_hosts=x86_64 ;;
-    i386) container_hosts=x86_64 ;;
+    i386) container_hosts="x86_64 aarch64 ppc ppc64" ;;
     loongarch64) container_hosts=x86_64 ;;
     m68k) container_hosts=x86_64 ;;
     microblaze) container_hosts=x86_64 ;;
     mips64el) container_hosts=x86_64 ;;
-    mips64) container_hosts=x86_64 ;;
+    mips64) container_hosts="x86_64 aarch64 ppc ppc64" ;;
     mipsel) container_hosts=x86_64 ;;
-    mips) container_hosts=x86_64 ;;
+    mips) container_hosts="x86_64 aarch64 ppc ppc64" ;;
     nios2) container_hosts=x86_64 ;;
-    ppc) container_hosts=x86_64 ;;
-    ppc64|ppc64le) container_hosts=x86_64 ;;
+    ppc) container_hosts="x86_64 aarch64 ppc ppc64" ;;
+    ppc64|ppc64le) container_hosts="x86_64 aarch64 ppc ppc64" ;;
     riscv64) container_hosts=x86_64 ;;
     s390x) container_hosts=x86_64 ;;
     sh4) container_hosts=x86_64 ;;
     sparc64) container_hosts=x86_64 ;;
     tricore) container_hosts=x86_64 ;;
-    x86_64) container_hosts="aarch64 ppc64el x86_64" ;;
+    x86_64) container_hosts="aarch64 ppc ppc64 ppc64el x86_64" ;;
     xtensa*) container_hosts=x86_64 ;;
   esac
 
